### PR TITLE
Manifest is outdated if both old and new manifests have digests

### PR DESCRIPTION
**Pull request by midnightercz**

### CODE REVIEW

**Review by querti — APPROVED** [review]

File: pubtools/_quay/push_docker.py [hunk 1]
```diff
@@ -819,11 +819,42 @@ def run(self) -> None:
...
+                outdated = True
+            elif s2_test:
+                d1 = bt1[0].v2s2_digest
+                d2 = bt2[0].v2s2_digest
+                name = "v2s2"
+                outdated = True
+            elif s1_test:
+                d1 = bt1[0].v2s1_digest  # noqa: F841
+                d2 = bt2[0].v2s1_digest  # noqa: F841
+                name = "v2s2"  # noqa: F841
```
Comment (emilyzheng) [inline pubtools/_quay/push_docker.py:851]:
```suggestion
                name = "v2s1"
```

File: pubtools/_quay/push_docker.py [hunk 2]
```diff
@@ -819,11 +819,42 @@ def run(self) -> None:
...
+                name = "v2s2"
+                outdated = True
+            elif s1_test:
+                d1 = bt1[0].v2s1_digest  # noqa: F841
+                d2 = bt2[0].v2s1_digest  # noqa: F841
+                name = "v2s1"
+                outdated = True
+            if outdated:
+                LOG.debug(
+                    "Marking manifest {bt1[0].tag} as outdated, {name} "
```
Comment (emilyzheng) [inline pubtools/_quay/push_docker.py:855]:
```suggestion
                    f"Marking tag {bt1[0].tag} as outdated, {name} "
```

File: pubtools/_quay/push_docker.py [hunk 3]
```diff
@@ -819,11 +819,42 @@ def run(self) -> None:
...
+                    "digests doesn't match {d1} != {d2}"
```
Comment (emilyzheng) [inline pubtools/_quay/push_docker.py:856]:
```suggestion
                    f"digests don't match {d1} != {d2}"
```

File: pubtools/_quay/push_docker.py (hunk 2 above)
Comment (midnightercz) [reply pubtools/_quay/push_docker.py:855]: But outdated is manifest, not tag

**Pull request merged**